### PR TITLE
chore: Reduce code duplication update request builder 

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Reduce code duplication for the update request builder.

### CODE REVIEW

File: test-packages/integration-tests/test/v2/change-detection-deep-update.spec.ts [hunk 1]
```diff
@@ -124,14 +124,14 @@ describe('deep-update and change detection', () => {
...
+    const request = await TestEntity.requestBuilder()
       .update(testEntity)
       .requiredFields(
         TestEntity.TO_SINGLE_LINK,
         TestEntity.COMPLEX_TYPE_PROPERTY
       )
       .execute(destination);
 
-    await expect(request).resolves.not.toThrow();
+    // await expect(request).resolves.not.toThrow();
```
Comment (jjtang1985) [inline test-packages/integration-tests/test/v2/change-detection-deep-update.spec.ts:135]: minor: left over

File: packages/core/src/odata-v4/request-builder/update-request-builder.ts [hunk 2]
```diff
@@ -41,37 +28,14 @@ export class UpdateRequestBuilderV4<EntityT extends EntityV4>
...
-    this.requestConfig.keys = oDataUriV4.getEntityKeys(
-      this._entity,
-      this._entityConstructor
+    super(
+      _entityConstructor,
+      _entity,
+      oDataUriV4,
+      entitySerializerV4,
+      extractODataEtagV4,
+      identity
```
Comment (jjtang1985) [inline packages/core/src/odata-v4/request-builder/update-request-builder.ts:37]: Question: why v4 is different in terms of the payload manipulation. Is it a bug or intended?

**Review by jjtang1985 — APPROVED** [review]
LGTM, I left a question.

File: test-packages/integration-tests/test/v2/change-detection-deep-update.spec.ts (hunk 1 above)
Comment (FrankEssenberger) [reply test-packages/integration-tests/test/v2/change-detection-deep-update.spec.ts:135]: oh good catch.

File: packages/core/src/odata-v4/request-builder/update-request-builder.ts (hunk 2 above)
Comment (FrankEssenberger) [reply packages/core/src/odata-v4/request-builder/update-request-builder.ts:37]: It is intended. For v2 the complex and nav props are removed. I do not know why. Can be overloaded with requiredFields.

**Review by jjtang1985 — APPROVED** [review]

**Pull request merged**